react-main/src/redux/login/actions.js

```diff
@@ -0,0 +1,63 @@
+import * as LoginService from '../../services/LoginService';
+import * as LocalStorageService from '../../services/LocalStorageService';
+
+export const actions = {
+  loginUser: 'LOGIN_USER',
+  loginSuccess: 'LOGIN_SUCCESS',
+  loginError: 'LOGIN_ERROR',
+  tokenExists: 'TOKEN_EXISTS',
+  tokenNotExists: 'TOKEN_NOT_EXISTS'
+};
+
+export const tokenExists = user => ({
+  type: actions.tokenExists,
+  message: `Welcome ${user.email}`,
+  session: true
+});
+
+export const tokenNotExists = () => ({
+  type: actions.tokenNotExists,
+  session: false
+});
+
+export const loginUser = () => ({
+  type: actions.loginUser
+});
+
+export const loginSuccess = () => ({
+  type: actions.loginSuccess,
+  message: 'Login correct!'
+});
+
+export const loginError = () => ({
+  type: actions.loginError,
+  message: 'Login error!'
+});
+
+const actionCreators = {
+  login: userAuth => async dispatch => {
+    dispatch(loginUser());
+    try {
```

Comment:
try/catch isn't necessary, since apisauce already handles errors.
LoginService.login should only return the response just as you receive it.
The action should check if response.ok, then success. Else, failure